test_extensions.py

```diff
@@ -36,3 +36,99 @@ def test_extensions_categories(base_url, selenium, count, category):
     assert 'sort=recommended%2Cusers' in selenium.current_url
     select = Select(category_results.filter_by_sort)
     assert 'Most Users' in select.first_selected_option.text
+
+
+@pytest.mark.nondestructive
+def test_extension_landing_header(base_url, selenium):
+    extensions = Extensions(selenium, base_url).open()
+    # checking that 'Extensions' is underlined in the header menu
+    assert 'Extensions' in extensions.header.is_active_link
+    assert 'Extensions' in extensions.title
+    assert 'Explore powerful tools and features' \
+           in extensions.header_summary
+
+
+@pytest.mark.nondestructive
+def test_recommended_extensions_shelf(base_url, selenium):
+    extensions = Extensions(selenium, base_url).open()
+    shelf_items = extensions.shelves.recommended_addons.list
+    assert 'Recommended extensions' in extensions.shelves.recommended_addons.card_header
+    # the following statements are checking that each shelf has four addons
+    # and each addon has a name, icon and number of users
+    assert len(shelf_items) == 4
+    for item in shelf_items:
+        assert item.name
+        assert item.addon_icon_preview.is_displayed()
+        assert item.addon_users_preview.is_displayed()
+
+
+@pytest.mark.desktop_only
+@pytest.mark.nondestructive
+def test_browse_more_recommended_extensions(base_url, selenium):
+    extensions = Extensions(selenium, base_url).open()
+    extensions.shelves.recommended_addons.browse_all
```

Comment:
I may have missed this in another PR but ```browse_all``` should be a function. But I can file an issue for that.